### PR TITLE
Add favor decision with comment moderation

### PR DESCRIPTION
**Pull request by hellcp-work**
Depends on #16008

Screenshot from 2024-04-18 14-14-54

### CODE REVIEW

File: src/api/app/models/decision_favored_moderate_comment.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  def description
+    'The moderator decided to favor the report and moderate the comment'
+  end
+
+  def self.display_name
+    'favored with comment moderation'
+  end
+
+  def self.display?(reportable)
+    return false unless reportable.class == Comment
```
Comment (danidoni) [inline src/api/app/models/decision_favored_moderate_comment.rb:14]:
```suggestion
    return false unless reportable.is_a?(Comment)
```

File: src/api/app/models/decision_favored_moderate_comment.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    return false if reportable.moderated?
+
+    true
+  end
+
+  def moderate_comment
+    reportable = reports.first.reportable
+    return unless reportable.class == Comment
```
Comment (danidoni) [inline src/api/app/models/decision_favored_moderate_comment.rb:22]:
```suggestion
    return unless reportable.is_a?(Comment)
```

**Comment by saraycp** [comment]
LGTM but I wonder if another naming can be more appropriate... What about `DecisionFavoredWithComemntModeration` or `DecisionFavoredWithUserDeletion`, etc.

What do you think @danidoni  @hellcp-work ?

**Comment by danidoni** [comment]
> LGTM but I wonder if another naming can be more appropriate... What about `DecisionFavoredWithCommentModeration` or `DecisionFavoredWithUserDeletion`, etc.
> 
> What do you think @danidoni @hellcp-work ?

I like it

**Review by saraycp — APPROVED** [review]

**Pull request merged**